app/controllers/session_controller.rb

```diff
@@ -53,6 +53,10 @@ def sso_provider(payload = nil)
         sso.admin = current_user.admin?
         sso.moderator = current_user.moderator?
         sso.groups = current_user.groups.pluck(:name)
+        base_url = Discourse.base_url
+        sso.avatar_url = base_url + current_user.avatar_template.gsub('{size}', '360')
```

Comment:
Is this change related to this PR? Also, there should be a test to verify this change.

Reply:
I think `avatar_url` is not provided in `sso_provider`, causing the existing code for sso client unable to grab avatar from sso provider. Please let me know if I'm wrong or it was added to another place. I also wondered if `avatar_url`, `profile_background_url` and `card_background_url` should be provided by the sso provider.

I added all of these test cases just now.

Reply:
Does this also work when uploads are stored on S3?